connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -800,6 +804,9 @@ message WithColumns {
   //
   // An exception is thrown when duplicated names are present in the mapping.
   repeated Expression.Alias aliases = 2;
+
+  // (Optional when `aliases` is set)
```

Comment:
Need more documentation.

For the curious reader. I did not remove `aliases` to retain backwards compatibility.